networking_calico/plugins/ml2/drivers/calico/policy.py

```diff
@@ -140,17 +140,16 @@ def _neutron_rule_to_etcd_rule(rule):
     if rule['protocol'] is None or rule['protocol'] == -1:
         pass
     elif rule['protocol'] == 'ipv6-icmp':
-        etcd_rule['protocol'] = 'ICMPv6'
+        etcd_rule['protocol'] = {'IPv6': 'ICMPv6'}[ethertype]
     elif rule['protocol'] == 'icmp':
-        etcd_rule['protocol'] = {'IPv4': 'ICMP',
-                                 'IPv6': 'ICMPv6'}[ethertype]
+        etcd_rule['protocol'] = {'IPv4': 'ICMP'}[ethertype]
```

Comment:
Yes, as per your later comment, we should leave this as it was before, for back-compatibility.